src/peer-id.js

```diff
@@ -12,17 +12,25 @@ async function downloadPeerIdFile({ awsClient, peerIdS3Region, peerIdS3Bucket, p
   return writeFile(peerIdJsonPath, contents)
 }
 
-async function getPeerId({ awsClient, peerIdS3Region, peerIdS3Bucket, peerIdJsonFile, peerIdJsonPath }) {
+async function getPeerId ({ awsClient, peerIdS3Region, peerIdS3Bucket, peerIdJsonFile, peerIdJsonPath }) {
+  if (!peerIdJsonPath) {
+    return createEd25519PeerId()
+  }
   if (peerIdS3Bucket) {
     await downloadPeerIdFile({ awsClient, peerIdS3Region, peerIdS3Bucket, peerIdJsonFile, peerIdJsonPath })
   }
 
   try {
     const peerIdJson = JSON.parse(await readFile(peerIdJsonPath, 'utf-8'))
-    return await PeerId.createFromJSON(peerIdJson)
-  } catch (e) {
-    return PeerId.create()
+    const { id, privKey, pubKey } = peerIdJson
+
+    const peer = await createFromJSON({ id, privKey, pubKey })
```

Comment:
the JSON is loaded, but it's not possible to connect to the peer from a "client"

```
/home/simone/code/ipfs/bitswap-peer-current/node_modules/p-some/index.js:31
                        reject(new AggregateError(errors));
                               ^

AggregateError: 
    Error: Error occurred during XX handshake: Error occurred while verifying signed payload: Peer ID doesn't match libp2p public key.
        at Noise.performXXHandshake (/home/simone/code/ipfs/bitswap-peer-current/node_modules/@web3-storage/libp2p-noise/dist/src/noise.js:163:19)
        at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
        at async Noise.performHandshake (/home/simone/code/ipfs/bitswap-peer-current/node_modules/@web3-storage/libp2p-noise/dist/src/noise.js:132:20)
        at async Noise.secureOutbound (/home/simone/code/ipfs/bitswap-peer-current/node_modules/@web3-storage/libp2p-noise/dist/src/noise.js:57:27)
        at async Upgrader._encryptOutbound (/home/simone/code/ipfs/bitswap-peer-current/node_modules/libp2p/src/upgrader.js:406:12)
        at async Upgrader.upgradeOutbound (/home/simone/code/ipfs/bitswap-peer-current/node_modules/libp2p/src/upgrader.js:176:11)
        at async ClassIsWrapper.dial (/home/simone/code/ipfs/bitswap-peer-current/node_modules/libp2p-websockets/src/index.js:54:18)
        at async TransportManager.dial (/home/simone/code/ipfs/bitswap-peer-current/node_modules/libp2p/src/transport-manager.js:111:14)
        at async /home/simone/code/ipfs/bitswap-peer-current/node_modules/libp2p/src/dialer/dial-request.js:71:18
        at async /home/simone/code/ipfs/bitswap-peer-current/node_modules/p-some/index.js:53:19
    at maybeSettle (/home/simone/code/ipfs/bitswap-peer-current/node_modules/p-some/index.js:31:11)
    at /home/simone/code/ipfs/bitswap-peer-current/node_modules/p-some/index.js:69:23
```